app/models/public_body.rb

```diff
@@ -921,8 +921,10 @@ def questions
   # will break unless we update index for every event for every
   # request linked to it
   def reindex_requested_from
-    return unless saved_change_to_attribute?(:url_name)
+    reindex_all_events if saved_change_to_attribute?(:url_name)
+  end
 
+  def reindex_all_events
```

Comment:
This is basically the same as `PublicBody#expire_requests`. Expiring also clears some caches, but perhaps we should just take the hit on additional processing here to remove some complexity? The new specs pass with `expire_requests if saved_change_to_attribute?(:url_name)`

Reply:
Yep I think it makes sense to switch to `expire_requests`